scripts/SREP/infra-node-resize/script.sh

```diff
@@ -0,0 +1,98 @@
+    yq -i 'del(.metadata.creationTimestamp) | del(.metadata.finalizer) | del(.metadata.resourceVersion) | del(.metadata.generation) | del(.metadata.selfLink) | del(.metadata.uid) | del(.status)' /tmp/cluster-infra.yaml
+    cat /tmp/cluster-infra.yaml 
+    CLOUD=$(oc -n "$NAMESPACE" get machinepool "$INFRA_MACHINEPOOL" -o json | jq -r '.spec.platform | keys_unsorted[0]')
+    yq -i ".metadata.name |= . + 2 | .spec.name |= . + 2 | .spec.platform.$CLOUD.type |= env(INSTANCE_SIZE)" /tmp/cluster-infra.yaml 
+}
+
+# create the temporary machinepool
+create_temp_machinepool() {
+    echo 'CREATING TEMPORARY MACHINEPOOL...'
+    oc -n "$NAMESPACE" create -f /tmp/cluster-infra.yaml
+    while [ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not correctly provisioned'
+        sleep 10
+    done
+    oc -n "$NAMESPACE" get machinepool
+}
+
+# delete the original machinepool
+delete_original_machinepool() {
+    echo 'DELETING THE ORIGINAL MACHINEPOOL...'
+    oc scale --replicas 0 machinepool "$INFRA_MACHINEPOOL" -n "$NAMESPACE"
+    oc -n "$NAMESPACE" get machinepool
+    while [ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not properly deleted'
+        sleep 10
+    done
+    oc -n "$NAMESPACE" delete machinepool "$INFRA_MACHINEPOOL"
+    echo 'Original machinepools which are not resized are deleted'
+    oc -n "$NAMESPACE" get machinepool
+}
+
+# update yaml file cluster 
+update_new_infra_yaml() {
+    echo 'UPDATING YAML FILE...'
+    yq eval '.metadata.name |= sub("2$"; "") | .spec.name |= sub("2$"; "")' -i /tmp/cluster-infra.yaml 
+    cat /tmp/cluster-infra.yaml
+}
+
+# create new machinepool
+create_new_machinepool() {
+    echo 'CREATING NEW MACHINEPOOL...'
+    oc -n "$NAMESPACE" create -f /tmp/cluster-infra.yaml
+    while [ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not correctly provisioned'
+        sleep 10
+    done
+}
+
+# delete temporary machinepool 
+delete_temp_machinepool() {
+    echo 'DELETING TEMPORARY MACHINEPOOL...'
+    oc scale --replicas 0 machinepool "${INFRA_MACHINEPOOL}"2 -n "$NAMESPACE"
+    oc -n "$NAMESPACE" get machinepool
+    while [ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not properly deleted'
+        sleep 10
+    done
+    oc -n "$NAMESPACE" delete machinepool "${INFRA_MACHINEPOOL}"2 
+    echo 'Temporary machinepools are deleted'
+}
+
+main() {
+  get_machinepool
+  update_cluster_infra_yaml
+  create_temp_machinepool
+  delete_original_machinepool
+  update_new_infra_yaml
+  create_new_machinepool
+  delete_temp_machinepool
```

Comment:
Let's have another function; something like `verify_machinepools` that will fetch (get) the existing, ready, and provisioned machinepools and verify it's current instance size matches the desired size and prints the same to the console: 

`MACHINE | INSTANCE SIZE | DESIRED INSTANCE SIZE`

More of a sanity check.